metaflow/plugins/kfp/kfp_utils.py

```diff
@@ -0,0 +1,171 @@
+    import kfp
+    import kfp_server_api
+except:
+    pass
+
+from metaflow.metaflow_config import KFP_RUN_URL_PREFIX, KFP_USER_DOMAIN
+from metaflow.plugins.kfp.kfp_constants import KFP_CLI_DEFAULT_SORT_BY
+from metaflow.util import get_username
+
+
+def _get_kfp_client():
+    kfp_client_user = get_username()
+    if KFP_USER_DOMAIN:
+        kfp_client_user += f"@{KFP_USER_DOMAIN}"
+    else:  # FIXME: The KFP_USER_DOMAIN value might not be available in cluster
+        kfp_client_user += "@zillowgroup.com"
+    return kfp.Client(userid=kfp_client_user)
+
+
+def get_pipeline_versions(
+    pipeline_name: str, sort_by=KFP_CLI_DEFAULT_SORT_BY, verbose=True
+) -> List[kfp_server_api.ApiPipeline]:
+    """Get kfp pipeline version by name. See get_pipeline_versions_by_id for details."""
+    client = _get_kfp_client()
+    pipeline_id = client.get_pipeline_id(pipeline_name)
+    return get_pipeline_versions_by_id(pipeline_id, sort_by=sort_by, verbose=verbose)
+
+
+def get_pipeline_versions_by_id(
+    pipeline_id: str, sort_by=KFP_CLI_DEFAULT_SORT_BY, verbose=True
+) -> List[kfp_server_api.ApiPipeline]:
+    """Get kfp pipeline version by id.
+    pipeline_id: Pipeline id in corresponding KFP server
+    sort_by: Can be format of “field_name”, “field_name asc” or “field_name desc”
+      (Example, “name asc” or “id desc”). Ascending by default.
+    verbose: Print pipeline info if True.
+      Default to True since this function is intended to be used interactively (human facing).
+      For similar reason `print` is used over logging.info to avoid silencing output by default
+      logging level.
+    """
+    client = _get_kfp_client()
+    versions = client.list_pipeline_versions(
+        pipeline_id=pipeline_id, sort_by=sort_by
+    ).versions
+
+    if verbose:
+        print(f"{len(versions)} versions found for pipeline {pipeline_id}.")
+        print(f"Versions sorted by {sort_by}.")
+
+        for version_count, version in enumerate(versions, start=1):
+            print(f"\n=== Version {version_count} ===")
+            for key, value in version.to_dict().items():
+                print(f"{key}: {value}")
+
+    return versions
+
+
+def trigger_flow(
+    pipeline_name: str,
+    triggerred_flow_name: str,
+    namespace: str,
+    experiment_name: str = None,
+    pipeline_parameters: dict = None,
+    pipeline_version: str = None,
+) -> str:
+    """Trigger KFP flow by pipeline name. See trigger_flow_by_id for more details."""
+
+    client = _get_kfp_client()
+
+    pipeline_id = client.get_pipeline_id(name=pipeline_name)
+    return trigger_flow_by_id(
+        pipeline_id=pipeline_id,
+        experiment_name=experiment_name,
+        namespace=namespace,
+        triggerred_flow_name=triggerred_flow_name,
+        pipeline_parameters=pipeline_parameters,
+        pipeline_version=pipeline_version,
+    )
+
+
+def trigger_flow_by_id(
+    pipeline_id: str,
+    triggerred_flow_name: str,
+    namespace: str,
```

Comment:
same as comment above: it'd be nice to put kubeflow_namespace to not confuse with Metaflow namespace